tests/unit/test_network_auth.py

```diff
@@ -12,26 +12,26 @@
     ["input_url", "url", "username", "password"],
     [
         (
-            "http://user%40email.com:password@example.com/path",
-            "http://example.com/path",
+            "https://user%40email.com:password@example.com/path",
+            "https://example.com/path",
```

Comment:
Why change from http to https? If there's a difference in behaviour, we should test both. If there isn't, this is unnecessary churn.

Reply:
True. Will revert those. I actually added tests below to make RFC compliant http/https distinction but http might remain here, no problem (the main issue was that my IDE complains about "insecure URLS" if there are http:// ones and It allows to change them to https:// easoly - which I did. But I am happy to remove that change.

Reply:
Fixed.